crates/rpc/rpc/src/eth/api/traits/pending_block.rs

```diff
@@ -148,4 +166,263 @@ pub trait LoadPendingBlock {
             Ok(Some(pending_block))
         }
     }
+
+    /// Assembles a [`Receipt`] for a transaction, based on its [`ExecutionResult`].
+    fn assemble_receipt(
+        &self,
+        tx: &TransactionSignedEcRecovered,
+        result: ExecutionResult,
+        cumulative_gas_used: u64,
+    ) -> Receipt {
+        Receipt {
+            tx_type: tx.tx_type(),
+            success: result.is_success(),
+            cumulative_gas_used,
+            logs: result.into_logs().into_iter().map(Into::into).collect(),
+            #[cfg(feature = "optimism")]
+            deposit_nonce: None,
+            #[cfg(feature = "optimism")]
+            deposit_receipt_version: None,
```

Comment:
we need to find a way to get rid of these features

Reply:
yeah...I know...

Reply:
they should be able to remove when optimism feature is entirely removed from the crate, as long as we have some dep for which we pull in optimism feature, likelihood that that crate pulls in `reth-primitives` with optimism feature, is very high - and we're screwed